internal/restic/node.go

```diff
@@ -299,8 +299,12 @@ func (node Node) createSymlinkAt(path string) error {
 	if runtime.GOOS == "windows" {
 		return nil
 	}
-	err := fs.Symlink(node.LinkTarget, path)
-	if err != nil {
+
+	if err := os.RemoveAll(path); err != nil && !errors.Is(err, os.ErrNotExist) {
```

Comment:
Wouldn't it be sufficient to use `Remove` in most cases? I'm not sure whether someone would expect restic to wipe a directory just to restore a directory in its place?

Reply:
My intention here was that I would expect to have the exact same state after a restore as I had at the time of the backup. Personally, I would be surprised if I had a symlink and after the restore there is still a directory instead of the sysmlink.
I would also be surprised if restic aborted with an error just because there was a directory in the restore destination that was a symlink at the time of the backup :shrug:.

Reply:
I'd prefer to play it safe, return an error in case there is a directory here and let the user sort it out.